Formatting changes (#696)
dtcenter/METplus: docs/Contributors_Guide/github_workflow.rst

GitHub Workflow
===============

How METplus releases are created
--------------------------------

The branching model employed by the METplus GitHub repository is similar to

> thatdescribed in

Reviewer: ```suggestion
that described in
```
missing space!

`A successful Git branching model <https://nvie.com/posts/a-successful-git-branching-model/>`_,
where new or updated code is created on a 'feature' branch that is based on
the `NCAR/METplus GitHub 'develop' branch <https://github.com/dtcenter/METplus/tree/develop>`_..

The feature branch is named after the corresponding GitHub issue:

| *feature_<Github Issue number>_<brief_description>*
|

When work is complete, the code in the feature branch is merged into the
develop branch.  When a release candidate for METplus has been determined,
then the develop branch is used to create a master_vx.y release of METplus,
which includes data tarballs for use in running use cases.

Sequence of Events - Contributing Code
--------------------------------------

*Pre-requisite:*

Set up a GitHub repository account if you do not already have one, and log
into your account.  For more information about GitHub accounts, please refer

> to the GitHub Documenation on `GitHub accounts <https://help.github.com/en/github/getting-started-with-github/signing-up-for-a-new-github-account>`_.

Reviewer: ```suggestion
to the GitHub Documentation on `GitHub accounts <https://help.github.com/en/github/getting-started-with-github/signing-up-for-a-new-github-account>`_.
```

| *feature_777_wrapper_xyz*
|

* Create your feature branch based off the develop branch

  .. code-block:: none

    git checkout develop

* Verify that you are currently working in the develop branch by running

  .. code-block:: none

    git branch

  You should see something like the following:

  .. code-block:: none

    * develop
    main_v3.1

  The asterisk, "*", indicates the currently active branch.

* At the command line, create and checkout the feature branch. For example:

  .. code-block:: none
		  
    git checkout -b feature_777_wrapper_xyz

  replacing *feature_777_wrapper_xyz* with your feature branch name.

* Verify that you are working in the correct branch by running:

  .. code-block:: none

    git branch

  You should see something like the following:

  .. code-block:: none
		  
    develop
    main_v3.1
    * feature_777_wrapper_xyz

  where the asterisk, "*", indicates which branch is currently in use/checked out.

Make changes to code in the feature branch
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

Users should make changes to their feature branch and commit changes to their
local repository.

> * Create code following the coding standards in the Coding Standards section of

Reviewer: Consider creating a link to the Coding Standards section

Author: Thanks!  Done!  Still need to commit this change.

Author: I have committed it.

  the Contributor's Guide.

* Provide some tests for your code using the pytest framework, provide user documentation
  describing what the code does, and provide any necessary data.

* Keep your fork in sync. While working, it is highly likely that changes are occurring in
  the original repository.  This may impact your work.  Regularly use the following commands
  to keep your fork in sync with the original repository.

  .. code-block:: none
		  
    git pull upstream develop
    git push origin develop

  The first command pulls changes from the original repository (the
  `METplus GitHub repository <https://github.com/dtcenter/METplus>`_ that you see when you
  run *git remote -v* and that you set to upstream in step 4 above).  The second command
  pushes those changes to your forked repository.

> 

Reviewer: Consider adding instructions to merge the local develop branch into the feature branch, i.e.

`git merge origin develop`

Author: Done!

Commit changes to feature branch
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

> * Commit any new files by running the following.  Perform this step only if this file is

Reviewer: ```suggestion
* Commit any new files by running the following.  Run the 'git add' command only if this file is
```
